tapir/coop/services/ResignMemberService.py

```diff
@@ -0,0 +1,43 @@
+import datetime
+
+from tapir.accounts.models import TapirUser
+from tapir.coop.models import ResignedMembership, ShareOwnership
+from tapir.shifts.models import ShiftAttendanceTemplate, ShiftAttendance
+
+from tapir.utils.shortcuts import get_timezone_aware_datetime
+
+class ResignMemberService:
+    @staticmethod
+    def update_shifts_and_shares(member: ResignedMembership):
+        tapir_user: TapirUser = getattr(member.share_owner, "user", None)
+        if not tapir_user:
+            print("Couldn't find an existing Tapir user.")
+            return
+        
+        new_end_date = member.cancellation_date
+        new_end_date = new_end_date.replace(day=31)
```

Comment:
This will not work for months with less than 31 days, i.e. today in June it throws exception:
```
web-1          |   File "/app/tapir/coop/services/ResignMemberService.py", line 18, in update_shifts_and_shares
web-1          |     new_end_date = new_end_date.replace(day=31)
web-1          |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1          | ValueError: day is out of range for month
```

Reply:
A placed the month before the day. Hopefully fixed.